src/Relude/Extra/Enum.hs

```diff
@@ -23,6 +26,21 @@ import Relude
 universe :: (Bounded a, Enum a) => [a]
 universe = [minBound .. maxBound]
 
+{- | Create a function that parses values of type @a@
+
+>>> data Color = Red | Green | Blue deriving (Show, Enum, Bounded)
+>>> parse = makeParser :: String -> Maybe Color
+>>> parse "Red"
+Just Red
+>>> parse "Black"
+Nothing
+-}
+makeParser :: forall a. (Bounded a, Enum a, Show a) => String -> Maybe a
+makeParser = \x -> Map.lookup x dict
```

Comment:
Is there any purpose to make it lambda function instead of having an argument on the right side?

Reply:
Oh, just in time: https://www.reddit.com/r/haskell/comments/93fe7i/functional_syntactic_sugar/e3csi2i/

Reply:
@vrom911 I'm not sure how it works... I always found this behavior confusing and difficult to understand. Yeah, that Reddit answer just in time! We can later just write benchmarks to see what's going on.

Reply:
@ghallak Yeah, I think if you add benchmarks for this function, it would be really great! It will help us all to understand performance of Haskell better. So we can test 3 cases:

1. This `makeParser` function.
2. `makeParser` written without lambda
3. Creating `HashMap` on top-level and reusing it.